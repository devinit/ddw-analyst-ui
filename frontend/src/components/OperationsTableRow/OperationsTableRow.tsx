import React, { FunctionComponent } from 'react';
import { DatasetRowItem } from '../DatasetRowItem';
import { OperationsTableRowActions } from '../OperationsTableRowActions';

export interface OperationsTableRowProps {
  count: number;
  name: string;
  updatedOn: string;
  isDraft: boolean;
  classNames?: string;
}

export const OperationsTableRow: FunctionComponent<OperationsTableRowProps> = (props) => {
  const renderActions = (): React.ReactNode =>
    React.Children.map(props.children, (child) => {
      if (React.isValidElement(child) && child.type === OperationsTableRowActions) {
        return child;
      }
    });

  return (
    <div className="dataset-row p-3 border-bottom">
      <div className="col-md-12">
        <DatasetRowItem text="Hola" addClass="dataset-row-title h4">
          <span className="badge badge-warning align-middle">Draft</span>
        </DatasetRowItem>
        <DatasetRowItem addClass="mb-2" text="Lets rock people we got this"></DatasetRowItem>

        <DatasetRowItem addClass="dataset-row-actions mb-1">{renderActions()}</DatasetRowItem>

        <DatasetRowItem addClass="h6 dataset-row-footer" text="Updated 2 days ago by">
          <span>Edwin P. Magezi</span> from <span className="text-uppercase text-danger">yolo</span>
        </DatasetRowItem>
      </div>
    </div>
  );
};
