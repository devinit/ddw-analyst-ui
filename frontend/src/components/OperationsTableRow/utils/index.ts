import axios from 'axios';
import * as localForage from 'localforage';
import { api, localForageKeys } from '../../../utils';

interface ServerResponse {
  id: number;
  indicator: string;
  indicator_acronym: string;
  source: string;
}

interface SourcesResponse {
  status: number;
  statusText: string;
  data: ServerResponse;
}

export const fetchSource = async (sourceId: number): Promise<SourcesResponse> => {
  const token = await localForage.getItem<string>(localForageKeys.API_KEY);

  return await axios.request({
    url: api.routes.SINGLE_SOURCE.replace('{sourceId}', `${sourceId}`),
    method: 'get',
    headers: {
      'Content-Type': 'application/json',
      Authorization: `token ${token}`,
    },
  });
};
