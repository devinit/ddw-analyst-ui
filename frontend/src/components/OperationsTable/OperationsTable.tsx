import * as React from 'react';
import { Table } from 'react-bootstrap';
import { OperationsTableRow } from '../OperationsTableRow';
import OperationsTableRowActions from '../OperationsTableRowActions';

export class OperationsTable extends React.Component {
  static Row = OperationsTableRow;
  static Actions = OperationsTableRowActions;

  render() {
    return (
      <Table responsive hover striped className="operations-table">
        <thead>
          <tr>
            <th>Name</th>
            <th>Updated On</th>
            <th>Status</th>
            <th className="text-right">Actions</th>
          </tr>
        </thead>
        <tbody>
          { this.renderRows() }
        </tbody>
      </Table>
    );
  }

  private renderRows() {
    return React.Children.map(this.props.children, child => {
      if (React.isValidElement(child) && child.type === OperationsTableRow) {
        return child;
      }
    });
  }
}
