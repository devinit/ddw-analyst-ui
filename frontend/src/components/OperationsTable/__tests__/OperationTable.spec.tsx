/**
 * @jest-environment jsdom
 */
import 'jest-styled-components';
import React from 'react';
import * as TestRenderer from 'react-test-renderer';
import { OperationsTableRow, OperationsTableRowProps } from '../../OperationsTableRow';
import { OperationsTable } from '../OperationsTable';

describe('OperationsTable', () => {
  test('renders correctly with the default props', () => {
    const renderer = TestRenderer.create(<OperationsTable />).toJSON();

    expect(renderer).toMatchSnapshot();
  });

  test('only renders OperationsTableRow children', () => {
    const props: OperationsTableRowProps = {
      count: 2,
      name: 'My Test Dataset',
      updatedOn: new Date('2020-01-01').toISOString(),
      isDraft: false,
    };
    const renderer = TestRenderer.create(
      <OperationsTable {...props}>
        <button type="button">Click Me</button>
        <OperationsTableRow {...props} />
      </OperationsTable>,
    ).toJSON();

    expect(renderer).toMatchSnapshot();
  });
});
