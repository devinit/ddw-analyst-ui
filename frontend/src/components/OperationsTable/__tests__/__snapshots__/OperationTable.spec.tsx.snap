// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OperationsTable only renders OperationsTableRow children 1`] = `
<div
  className="card dataset-list"
>
  <div
    className="card-body p-0"
  >
    <div
      className="dataset-row p-3 border-bottom"
    >
      <div
        className="col-md-12"
      >
        <div
          className="dataset-row-title h4"
        >
          My Test Dataset
           
        </div>
        <div
          className="mb-2"
        >
           
        </div>
        <div
          className="dataset-row-actions mb-1"
        >
           
        </div>
        <div
          className="h6 dataset-row-footer"
        >
          Updated
           
           by 
          <span>
             
             
          </span>
          from 
          <span
            className="text-uppercase text-danger"
          >
            
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`OperationsTable renders correctly with the default props 1`] = `
<div
  className="card dataset-list"
>
  <div
    className="card-body p-0"
  />
</div>
`;
